tasks.py

```diff
@@ -0,0 +1,49 @@
+from invoke import task
+import os
+
+
+@task
+def install_dependencies(c):
+    c.run("pip install -r requirements.txt")
+
+
+@task
+def check_format(c):
+    c.run("black --check python/*.py")
+
+
+@task
+def lint(c):
+    c.run("flake8 .")
+
+
+@task
+def build_dev_docker(c):
+    c.run("docker build -t iati_partner_search .")
+
+
+@task
+def build_docker(c):
+    tag = os.environ["TRAVIS_BUILD_NUMBER"] if "TRAVIS_BUILD_NUMBER" in os.environ else "latest"
+    c.run(f"docker build -t rabshab/iati-partner-search-app:{tag} -f app.Dockerfile .")
```

Comment:
Should we use the `TRAVIS_BUILD_NUMBER` here? I'm wondering if it's possible to use the git commit tag, so that it could ostensibly be built independently of travis. There might be some logic that I'm missing here though.

```suggestion
    c.run(f"docker build -t rabshab/iati-partner-search-app:$(git rev-parse --short HEAD) -f app.Dockerfile .")
```

Reply:
For this one it uses the travis build number if its present but if not, and its being done locally, it uses "latest". I didn't add this check for the `push_docker` task as ideally you wouldn't be doing that locally.

I think the way to make this a bit clearer would just be to add a new task `build_docker_travis` that uses travis env vars

I quite like using the travis build number for tagging stuff as it is a unique, incremental number. Means that you get nice sorting stuff for free as the latest build will always be the one with the biggest number.

Reply:
Cool! Sounds good - hadn't considered what Travis gives you with its number. I like the idea of a `build_docker_travis` task in the future.